packages/scene-composer/src/components/StateManager.tsx

```diff
@@ -245,22 +246,21 @@ const StateManager: React.FC<SceneComposerInternalProps> = ({
       return;
     }
 
-    if (sceneInfo && sceneMetadataModule && sceneInfo.capabilities?.includes(SceneCapabilities.DYNAMIC_SCENE)) {
-      // Fetch scene level metadata for dynamic scene
-      const rootId = sceneInfo.sceneMetadata?.[SceneMetadataMapKeys.SCENE_ROOT_ENTITY_ID];
-      if (!rootId) {
-        setLoadSceneError(new Error('Failed to get scene root entity id'));
-        return;
+    if (sceneRootEntityId && sceneInfo && sceneMetadataModule) {
+      if (sceneInfo.contentLocation && sceneInfo.contentLocation.length > 0) {
+        setSceneContentUri(sceneInfo.contentLocation!);
       }
 
       sceneMetadataModule
-        .getSceneEntity({ entityId: rootId })
+        .getSceneEntity({ entityId: sceneRootEntityId })
         .then((res) => {
-          const document = parseSceneCompFromEntity(res);
-          if (!document) {
-            throw new Error('Failed to parse scene metadata');
+          if (res?.components?.length) {
+            const document = parseSceneCompFromEntity(res);
```

Comment:
So right now this executes and doesn't fail out until if(res?.components?.length)  realizes we have no scene component?  Does that still avoid the early return  on 269?

Reply:
Original code assume there is scene Node entity.  It's hard to remove completely. So, the extra check is to ensure the sceneInfo is from scene.json, not from a sceneNode. I added this line after saw crash issue when updating a property of scene object in the composer.

Reply:
Okay, looks like the setSceneContentUri on line 251 ensures we have a way to load scene content if there is no scene component on the scene entity.